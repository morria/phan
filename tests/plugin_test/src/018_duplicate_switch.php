<?php
// Phan's DuplicateKeyPlugin can analyze switch statements for duplicate cases, similar to the way duplicate keys are analyzed.

const Z = 1;
const Y = 1;
const A = true;
const B = false;

class NSwitch {
    const XYZ = 1;
    const TRUE = true;
    const FALSE = false;
}

$x = rand() % 10;
switch($x) {
case Y: return 1;
case Z: return 2;
case NSwitch::XYZ: return 3;
case NSwitch::TRUE: return true;
case NSwitch::FALSE: return false;
case A: return true;
case B: return false;
case false: return 11;
case true: return 13;
case null: return 14;
case '': return 15;
case '(test literal)': return 15;
case '(test literal)': return 16;
// Could analyze these later, but not doing that yet. For now, just check that this doesn't warn.
case __FILE__: return 15;
case __LINE__: return 15;
case __LINE__ - 1: return 15;
case '': return 15;
case null: return 14;
default: return 'default';
};
