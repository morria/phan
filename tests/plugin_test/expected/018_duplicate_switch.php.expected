src/018_duplicate_switch.php:18 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(1) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:19 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(1) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:22 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(true) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:23 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(false) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:24 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(false) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:25 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(true) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:29 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case('(test literal)') detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:33 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(32) detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:34 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case('') detected in switch statement - the later entry will be ignored.
src/018_duplicate_switch.php:35 PhanPluginDuplicateSwitchCase Duplicate/Equivalent switch case(null) detected in switch statement - the later entry will be ignored.
